### PR TITLE
Avoid walls first pass

### PR DESCRIPTION
**Pull request by Feesh7**
Very basic implementation of wall avoidance for the camera. Will be upgraded significantly over time, just needed something to start with so people can walk around indoor maps.

### CODE REVIEW

**Comment by Feesh7** [comment]
:cold_sweat:

**Comment by Feesh7** [comment]
ᕙ(˵ ಠ ਊ ಠ ˵)ᕗ

File: Assets/Scripts/CameraController.cs [hunk 1]
```diff
@@ -99,13 +107,20 @@ public void ThirdPersonJumpCallback()
     // LateUpdate is called after update. Ensures we are operating on the latest transform changes.
     void LateUpdate () {
         UpdateCameraAngles();
-	}
+    }
```
Comment (nayrpg) [inline Assets/Scripts/CameraController.cs]: I love how you pretty you made your code :heart_eyes:

**Comment by Feesh7** [comment]
ᕕ( ՞ ᗜ ՞ )ᕗ

**Comment by Feesh7** [comment]
(╯°□°)╯︵ ┻━┻

**Comment by Feesh7** [comment]
[̲̅$̲̅(̲̅ ͡° ͜ʖ ͡°̲̅)̲̅$̲̅]

File: Assets/Scripts/CameraController.cs [hunk 2]
```diff
@@ -25,6 +25,10 @@ public class CameraController : MonoBehaviour {
     private CameraMovementFunction handleCameraMove;
     private CameraMovementFunction handlePlayerRotate;
 
+    // Timers
+    private float WallHitTimeDelta;
+    private float WallHitGracePeriod;
```
Comment (nayrpg) [inline Assets/Scripts/CameraController.cs]: How did you know I wanted a WallHitGracePeriod :bouquet:

**Comment by Feesh7** [comment]
ヽ༼◥▶ل͜◀◤༽ﾉ

**Comment by Feesh7** [comment]
(∩ ͡° ͜ʖ ͡°)⊃━☆─=≡Σ((( つ◕ل͜◕)つ

File: Assets/Scripts/CameraController.cs [hunk 3]
```diff
@@ -151,6 +166,28 @@ private void ThirdPersonCameraMove()
             mouseAccumlator.x = Mathf.LerpAngle(mouseAccumlator.x, yaw, 0.1f);
             mouseAccumlator.y = Mathf.LerpAngle(mouseAccumlator.y, adjusted_pitch, 0.1f);
         }
+        AvoidWalls();
+    }
+
+    private void AvoidWalls()
```
Comment (nayrpg) [inline Assets/Scripts/CameraController.cs]: I never want to avoid your code :hugs:

**Pull request merged**